### PR TITLE
Pass-Supported-Platform

### PR DESCRIPTION
**Pull request by ahmadiesa-abu**
this PR address platform issue in case of new manylinux wheel naming convention [pep600 concatenate]

### CODE REVIEW

**Review by EarthmanT — DISMISSED** [review]

File: wagon.py [hunk 1]
```diff
@@ -746,6 +746,18 @@ def create(source,
...
+        # example :
+        # >>> platform='manylinux_2_5_x86_64.manylinux1_x86_64'
+        # >>> while 'manylinux' in platform.partition('_')[2]:
+        # ...     platform=platform.partition('_')[2]
+        # >>> platform
+        # '64.manylinux1_x86_64'
+        # >>> platform.partition('_')
+        # ('64.manylinux1', '_', 'x86_64')
+        while 'manylinux' in platform.partition('_')[2]:
+            platform=platform.partition('_')[2]
```
Comment (tehasdf) [inline wagon.py:760]:
```suggestion
            platform = platform.partition('_')[2]
```
huh we dont have flake8 in ci here, and apparently ci doesnt work anyway

**Review by tehasdf — CHANGES_REQUESTED** [review]
this wont work because in the simple case of platform = `manylinux_2_5_x86_64` (which is a pep600 alias for `manylinux1_x86_64`), the platform here ends up being `linux_2_5_x86_64`, which isn't valid (`linux_x86_64` is)

**Comment by tehasdf** [comment]
all in all the best idea is probably to just drop the 5.1-compat hack, because passing in the `--supported-platform` correctly means that plugins that need to be 5.1-compatible, can just be built with the override passed in

**Comment by ahmadiesa-abu** [comment]
> this wont work because in the simple case of platform = `manylinux_2_5_x86_64` (which is a pep600 alias for `manylinux1_x86_64`), the platform here ends up being `linux_2_5_x86_64`, which isn't valid (`linux_x86_64` is)

I have added another snippet that handles these extreme cases

**Comment by ahmadiesa-abu** [comment]
> all in all the best idea is probably to just drop the 5.1-compat hack, because passing in the `--supported-platform` correctly means that plugins that need to be 5.1-compatible, can just be built with the override passed in

I agree , that is why we fixed the create function to pass in `supported-platform` , but we also need to be able to handle it , if the user doesn't pass it also

**Review by tehasdf — APPROVED** [review]

**Review by EarthmanT — APPROVED** [review]

**Pull request merged**